source/MRViewer/MRRenderPointsObject.cpp

```diff
@@ -416,18 +416,22 @@ RenderBufferRef<unsigned> RenderPointsObject::loadVertSelectionTextureBuffer_()
 
     const auto& points = objPoints_->pointCloud();
     const auto step = objPoints_->getRenderDiscretization();
-    const auto numV = ( points->validPoints.find_last() + 1 ) / int( step );
+    const int num = points->validPoints.find_last() + 1;
+    const auto numV = num / int( step );
     auto size = numV / 32 + 1;
     vertSelectionTextureSize_ = calcTextureRes( size, maxTexSize_ );
     assert( vertSelectionTextureSize_.x * vertSelectionTextureSize_.y >= size );
     auto buffer = glBuffer.prepareBuffer<unsigned>( vertSelectionTextureSize_.x * vertSelectionTextureSize_.y );
 
-    const auto& selection = objPoints_->getSelectedPoints().m_bits;
-    const unsigned* selectionData = (unsigned*) selection.data();
+    const auto& selectedPoints = objPoints_->getSelectedPoints();
+    const size_t selectionSize = selectedPoints.m_bits.size();
+    
+    const unsigned* selectionData = ( unsigned* )selectedPoints.m_bits.data();    
+
     ParallelFor( 0, ( int )buffer.size(), [&]( int r )
     {
         auto& block = buffer[r];
```

Comment:
Suggestion:
`block = 0;` in the beginning
Leave only bit sets to 1: `block |= 1 << bit;`